src/rest/directive.rs

```diff
@@ -56,3 +56,135 @@ impl TryFrom<&Directive> for Rest {
+                @rest(method: DELETE, path: "/foo/$a", query: {b: $b, c: $c, d: $d}) {
+                    value
+                }"#.to_string() =>
+            Rest::default()
+                .path("/foo/$a".to_string())
+                .method(Some(Method::DELETE))
+                .query(
+                    btreemap! { "b".to_string() => "b".to_string(), "c".to_string() => "c".to_string(), "d".to_string() => "d".to_string() },
+                ),
+
+            // PATCH method
+            r#"query ($a: Int, $b: String, $c: Boolean, $d: Float, $v: String)
+                        @rest(method: PATCH, path: "/foo/$a", query: {b: $b, c: $c, d: $d}, body: $v) {
+                            value
+                        }"#.to_string() =>
+            Rest::default()
+                .path("/foo/$a".to_string())
+                .method(Some(Method::PATCH))
+                .query(
+                    btreemap! { "b".to_string() => "b".to_string(), "c".to_string() => "c".to_string(), "d".to_string() => "d".to_string() },
+                )
+                .body(Some("v".to_string())),
+
+
+            // HEAD method
+            r#"query ($a: Int, $b: String, $c: Boolean, $d: Float)
+                        @rest(method: HEAD, path: "/foo/$a", query: {b: $b, c: $c, d: $d}) {
+                            value
+                        }"#.to_string() =>
+            Rest::default()
+                .path("/foo/$a".to_string())
+                .method(Some(Method::HEAD))
+                .query(
+                    btreemap! { "b".to_string() => "b".to_string(), "c".to_string() => "c".to_string(), "d".to_string() => "d".to_string() },
+                ),
+
+            // OPTIONS method
+            r#"query ($a: Int, $b: String, $c: Boolean, $d: Float)
+                @rest(method: OPTIONS, path: "/foo/$a", query: {b: $b, c: $c, d: $d}) {
+                    value
+                }"#.to_string() =>
+            Rest::default()
+                .path("/foo/$a".to_string())
+                .method(Some(Method::OPTIONS))
+                .query(
+                    btreemap! { "b".to_string() => "b".to_string(), "c".to_string() => "c".to_string(), "d".to_string() => "d".to_string() },
+                ),
+
+
+
+        }
+    }
+
+    fn query_to_directive(query: &str) -> Directive {
+        async_graphql::parser::parse_query(query)
+            .unwrap()
+            .operations
+            .iter()
+            .next()
+            .unwrap()
+            .1
+            .node
+            .directives
+            .first()
+            .unwrap()
+            .node
+            .clone()
+    }
+
+    #[test]
+    fn test_directive_to_rest() {
+        for (query, expected_rest) in queries_map() {
+            let directive = query_to_directive(&query);
+            let actual = Rest::try_from(&directive).unwrap();
+            assert_eq!(actual, expected_rest);
+        }
+    }
+}
```

Comment:
Error handling is irrelevant because this is a test case, panics are informative enough 
And the boilerplate is explicit so that tests can be written with flexibility